In [3]:
import tensorflow_datasets as tfds
print(len(tfds.list_builders()))
print(tfds.list_builders()[:5])

/Users/yujin/anaconda3/envs/machine_learning/lib/python3.9/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm
2024-08-04 15:02:42.777121: W tensorflow/tsl/platform/cloud/google_auth_provider.cc:184] All attempts to get a Google authentication bearer token failed, returning an empty token. Retrieving token from files failed with "NOT_FOUND: Could not locate the credentials file.". Retrieving token from GCE failed with "FAILED_PRECONDITION: Error executing an HTTP request: libcurl code 6 meaning 'Couldn't resolve host name', error details: Could not resolve host: metadata.google.internal".


1226
['abstract_reasoning', 'accentdb', 'aeslc', 'aflw2k3d', 'ag_news_subset']


현재 1226개의 데이터셋이 사용 가능하다,
처음 5개 데이터셋의 리스트는 ['abstract_reasoning', 'accentdb', 'aeslc', 'aflw2k3d', 'ag_news_subset'] 이다.

In [5]:
celeba_bldr = tfds.builder('celeb_a')
print(celeba_bldr.info.features)

FeaturesDict({
    'attributes': FeaturesDict({
        '5_o_Clock_Shadow': bool,
        'Arched_Eyebrows': bool,
        'Attractive': bool,
        'Bags_Under_Eyes': bool,
        'Bald': bool,
        'Bangs': bool,
        'Big_Lips': bool,
        'Big_Nose': bool,
        'Black_Hair': bool,
        'Blond_Hair': bool,
        'Blurry': bool,
        'Brown_Hair': bool,
        'Bushy_Eyebrows': bool,
        'Chubby': bool,
        'Double_Chin': bool,
        'Eyeglasses': bool,
        'Goatee': bool,
        'Gray_Hair': bool,
        'Heavy_Makeup': bool,
        'High_Cheekbones': bool,
        'Male': bool,
        'Mouth_Slightly_Open': bool,
        'Mustache': bool,
        'Narrow_Eyes': bool,
        'No_Beard': bool,
        'Oval_Face': bool,
        'Pale_Skin': bool,
        'Pointy_Nose': bool,
        'Receding_Hairline': bool,
        'Rosy_Cheeks': bool,
        'Sideburns': bool,
        'Smiling': bool,
        'Straight_Hair': bool,
        'Wavy_Hair': b

In [6]:
print(celeba_bldr.info.features['image'])

Image(shape=(218, 178, 3), dtype=uint8)


In [7]:
print(celeba_bldr.info.citation)

@inproceedings{conf/iccv/LiuLWT15,
  added-at = {2018-10-09T00:00:00.000+0200},
  author = {Liu, Ziwei and Luo, Ping and Wang, Xiaogang and Tang, Xiaoou},
  biburl = {https://www.bibsonomy.org/bibtex/250e4959be61db325d2f02c1d8cd7bfbb/dblp},
  booktitle = {ICCV},
  crossref = {conf/iccv/2015},
  ee = {http://doi.ieeecomputersociety.org/10.1109/ICCV.2015.425},
  interhash = {3f735aaa11957e73914bbe2ca9d5e702},
  intrahash = {50e4959be61db325d2f02c1d8cd7bfbb},
  isbn = {978-1-4673-8391-2},
  keywords = {dblp},
  pages = {3730-3738},
  publisher = {IEEE Computer Society},
  timestamp = {2018-10-11T11:43:28.000+0200},
  title = {Deep Learning Face Attributes in the Wild.},
  url = {http://dblp.uni-trier.de/db/conf/iccv/iccv2015.html#LiuLWT15},
  year = 2015
}


데이터셋 설명
image, landmarks, attributes 세 개의 키를 가진 딕셔너리
shape=(218, 178, 3) 3차원 데이터셋
'image'키는 유명인사의 얼굴 이미지를 담고있다.
'landmarks'키는 얼굴에서 추출한 위치로 구성된 딕셔너리 (ex. 눈,코,입 위치)
'attributes' 키는 이미지에 있는 사람의 얼굴속성 (ex. 표정, 화장, 머리카락)

In [9]:
celeba_bldr.download_and_prepare()
# 데이터셋을 위해 지정된 파일에 데이터셋을 저장한다.

Dl Completed...:  60%|██████    | 3/5 [00:14<00:09,  4.84s/ url]


NonMatchingChecksumError: Artifact https://drive.google.com/uc?export=download&id=0B7EVK8r0v71pZjFTYXZWM3FlRnM, downloaded to /Users/yujin/tensorflow_datasets/downloads/ucexport_download_id_0B7EVK8r0v71pZjFTYXZWM3FlDDaXUAQO8EGH_a7VqGNLRtW52mva1LzDrb-V723OQN8.tmp.2e5a7531dc6d4824af89dd6c0727d0a7/download, has wrong checksum:
* Expected: UrlInfo(size=1.34 GiB, checksum='46fb89443c578308acf364d7d379fe1b9efb793042c0af734b6112e4fd3a8c74', filename='img_align_celeba.zip')
* Got: UrlInfo(size=2.37 KiB, checksum='66ec5670a65e5754c2e6e0ed6c96c97f008ee4ccb3557e4683b7b3b5e65fefef', filename='download')
To debug, see: https://www.tensorflow.org/datasets/overview#fixing_nonmatchingchecksumerror

In [10]:
import tensorflow_datasets as tfds
import pathlib

# 수동으로 다운로드한 파일 경로 설정
download_dir = pathlib.Path('~/tensorflow_datasets/manual_dir')
img_align_celeba_zip = download_dir / 'img_align_celeba.zip'

# 다운로드한 파일이 있는지 확인
assert img_align_celeba_zip.exists(), "파일이 존재하지 않습니다. 수동으로 다운로드 후 지정된 경로에 저장하세요."

# 데이터셋 빌더 설정
celeba_bldr = tfds.builder('celeb_a')
celeba_bldr.download_and_prepare(download_dir=download_dir)


AssertionError: 파일이 존재하지 않습니다. 수동으로 다운로드 후 지정된 경로에 저장하세요.